contents/android/IntroToAndroid.md

```diff
@@ -0,0 +1,108 @@
+
+([Image](https://developer.android.com/guide/components/images/activity_lifecycle.png) from Google)
+
+Activity instances in an app will transition through different stages in their `lifecycle`. An activity that has just been created will be in the `Created` stage. When the app prepares for the activity to enter the foreground and be displayed to the user, it enters the `Started` stage. The UI could be initialised at this stage. When the activity comes to the foreground and can be viewed by the user, it enters the `Resumed` stage. More details about the other stages can be found in the picture above or at the [Activity Lifecycle](https://developer.android.com/guide/components/activities/activity-lifecycle.html) page.
+
+The `Activity` class implements callbacks that will be invoked as the activity enters a new state. These callbacks can be overridden in your activity class that inherits from `Activity` so that your app can respond to transitions between states.
+
+The following code is an example of overriding the `onCreate` callback. It calls the superclass method first and then declares the user interface which is defined in an XML layout file `R.layout.activity_main`.
+
+```
+public class MainActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+    }
+```
+
+### 2.4 MVVM and Architecture Components
+At Google I/O 2017, [Architecture Components](https://developer.android.com/topic/libraries/architecture/index.html) were introduced to help developers solve common issues by following best practices and implementing recommended architecture.
+
+#### LiveData
+Some examples of useful components for developers include the `LiveData` observable data holder class. Android apps usually use [RxJava](https://github.com/ReactiveX/RxJava/wiki/How-To-Use-RxJava) to do [reactive programming](https://blog.mindorks.com/rxjava-anatomy-what-is-rxjava-how-rxjava-is-designed-and-how-rxjava-works-d357b3aca586), which extends the `observable pattern` of Software Engineering. However, now with `LiveData`, we are able to create observables that are `lifecycle-aware`, only updating app components that are active. This prevents memory leaks and crashes. For more information, see the [LiveData page](https://developer.android.com/topic/libraries/architecture/livedata.html).
+
+#### ViewModel
+Another very useful component is the `ViewModel` component. It manages and store UI data in a `lifecycle-aware` way, allowing data to be retained during configuration changes such as screen rotations. Using normal UI controllers require data to be saved and then restored, but this is not practical for large amounts of data. For more information, see the [ViewModel page](https://developer.android.com/topic/libraries/architecture/viewmodel.html).
```

Comment:
`Using normal UI controllers require data to be saved and then restored, but this is not practical for large amounts of data.` 

Might be good to elaborate on what you mean by `not practical`. From what I remember, saving the state of activities & fragments was a pain point because there's a need to code it manually. (but maybe things have changed recently so I'm not so sure about this, maybe you were referring to something else)